.grit/patterns/valibot_v03.md

```diff
@@ -0,0 +1,105 @@
+# Valibot v0.31 Upgrade
+
+In v0.31, the API was changed to use a `v.pipe` method.
+
+```grit
+
+pattern vb_renames() {
+  or {
+    `custom` => `check`,
+    `BaseSchema` => `GenericSchema`,
+    `Input` => `InferInput`,
+    `Output` => `InferOutput`,
+    `SchemaConfig` => `Config`,
+    `special` => `custom`,
+    `toCustom` => `transform`,
+    `toTrimmed` => `trim`,
+    `toTrimmedEnd` => `trimEnd`,
+    `toTrimmedStart` => `trimStart`
+  }
+}
+
+pattern vb_pipe() {
+  or {
+    `v.string([$args])` => `v.pipe(v.string(), $args)`,
+  }
+}
+
+pattern vb_coerce() {
```

Comment:
I would change it to something like this:

```
pattern vb_coerce() {
  or {
    `v.coerce($schema, $fn)` => `v.pipe(v.unknown(), v.transform($fn))`
  }
}
```

Reply:
Sounds good!